shell/platform/darwin/ios/framework/Source/FlutterViewController.mm

```diff
@@ -897,8 +927,23 @@ - (void)dispatchTouches:(NSSet*)touches
     return;
   }
 
+  // When UIApplicationSupportsIndirectInputEvents is set in Info.plist, mouse clicks will be
+  // received with touch.type of UITouchTypeIndirectPointer, which will be correctly translated to
+  // flutter::PointerData::DeviceKind::kMouse. But since we use an identifier unique to the touch
+  // as its device identifier, after the mouse click ends, the device needs to be removed from the
+  // framework. If the device is not removed, the released click will keep triggering hover events
+  // on MouseRegions, with no way to remove it. This didn't happen previously because the framework
+  // only does this hover logic when device kind is flutter::PointerData::DeviceKind::kMouse.
+  size_t touches_to_remove_count = 0;
+  for (UITouch* touch in touches) {
+    if (touch.phase == UITouchPhaseEnded || touch.phase == UITouchPhaseCancelled) {
```

Comment:
same here check for flutter_state_is_added?

Reply:
See above comment, `flutter_state_is_added` is only for the mouse hover cursor.